### ISSUE TITLE
Nanoserver images not working, missing netapi32.dll

### ISSUE THREAD

**Issue by TheJayMann** [lucaslorentz/caddy-docker-proxy#333]
It appears that, sometime between the last time I used this project for Windows containers and now (I'd assume most likely during the upgrade to Caddy v2), there is now a dependency on netapi32.dll which does not exist in Nanoserver container images.  The end result is that the images no longer with Windows containers.

One solution to this would be to switch from using Nanoserver to using Server Core, though this has the issue of adding approximately 2 GB to the container image size.  Alternatively, another solution I've seen from Stefan Scherer is to pull both the Nanoserver and Server Core images, then copy the netapi32.dll file from the Server Core image, as shown in https://github.com/StefanScherer/dockerfiles-windows/blob/main/traefik/Dockerfile.1809#L3.

**Comment by TheJayMann** [comment]
Given that I've now had time to go back and further work on this, I have been able to get the windows images to work using a custom derived image using the following dockerfile.
```Dockerfile
FROM mcr.microsoft.com/windows/servercore:ltsc2019 as core
FROM lucaslorentz/caddy-docker-proxy:ci-nanoserver-1809
COPY --from=core /windows/system32/netapi32.dll /windows/system32/netapi32.dll
```
golang/go#21867 and microsoft/Windows-Containers#72 both mention the missing dll file.  Until resolved, either by Windows, golang, or caddy, in order for caddy-docker-proxy to work in a container, either the netapi32.dll file will need to be copied in the nanoserver image (which might possibly have redistribution issue), or the image will need to be based on ServerCore (which increases the image size significantly).

Further, I would like to note, while trying to get things working on a Windows 2019 swarm mode cluster, the service accessing the docker engine pipe (either the controller, or the standalone) either needs to run as the ContainerAdministrator user, or use HyperV isolation, and in order to use the routing mesh to map the ports of the caddy server or standalone, the service needs to use HyperV isolation, as I was not able to get the ingress routing to work when using process isolation and a not default network.  Trying to use host port mapping in global deployment mode, I believe I got it to work sometimes, with no real consistent results.
